ci_tools/process_results.py

```diff
@@ -10,25 +10,38 @@
                         help='Github step summary')
 args = parser.parse_args()
 
+all_codes = ['GL01', 'GL02', 'GL03', 'GL05', 'GL06', 'GL07', 'GL08', 'GL09',
+             'GL10', 'SS01', 'SS02', 'SS03', 'SS04', 'SS05', 'SS06', 'ES01',
+             'PR01', 'PR02', 'PR03', 'PR04', 'PR05', 'PR06', 'PR07', 'PR08',
+             'PR09', 'PR10', 'RT01', 'RT02', 'RT03', 'RT04', 'RT05', 'YD01',
+             'SA01', 'SA02', 'SA03', 'SA04', 'EX01']
+
 warning_codes = ['EX01', 'SA01']
```

Comment:
Right now `all_codes` contains both error and warning codes, and after specifying `warning_codes` you know that the remaining codes must be errors.

Instead I find it clearer to define two disjoint sets, `error_codes` and `warning_codes`. You can use an assertion to make sure there is no duplicate:

```python
warning_codes = {'EX01', 'SA01'}
error_codes = {'GL01', ..., 'SA03'}

assert warning_codes & error_codes == set()
```
And later in the loop you can write clearly:

```python
if code in warning_codes:
    ...
elif code in error_codes:
    ...
else:
    parsing_errors.append(line)
    # no need for 'continue' statement
```

Reply:
got it! i will fix it!